tests/test_replicate.py

```diff
@@ -268,16 +326,124 @@ def test_call_streamming_method_called():
+    urls_metadata = {
+        "s3://tcga-open/uuid1/filname1": {},
+        "gs://gdc-tcga-open/uuid1/filname1": {},
+    }
+    url = "s3://tcga-controlled/uuid1/filname1"
+
+    doc = MockIndexDRecord(urls=urls, urls_metadata=urls_metadata)
+    doc, modified = indexd_utils._remove_changed_url(doc, url)
+    assert modified == True
+    assert doc.urls == ["gs://gdc-tcga-open/uuid1/filname1"]
+    assert doc.urls_metadata == {"gs://gdc-tcga-open/uuid1/filname1": {}}
+
+    urls = ["s3://ccle-open-access/uuid1/filname1", "gs://gdc-ccle-open/uuid1/filname1"]
+    urls_metadata = {
+        "s3://ccle-open-access/uuid1/filname1": {},
+        "gs://gdc-ccle-open/uuid1/filname1": {},
+    }
+    url = "s3://gdc-ccle-controlled/uuid1/filname1"
+
+    doc = MockIndexDRecord(urls=urls, urls_metadata=urls_metadata)
+    doc, modified = indexd_utils._remove_changed_url(doc, url)
+    assert modified
+    assert doc.urls == ["gs://gdc-ccle-open/uuid1/filname1"]
+    assert doc.urls_metadata == {"gs://gdc-ccle-open/uuid1/filname1": {}}
+
+
+def test_is_changed_acl_object():
+    """
+    Test that acl object is changed
+    :return:
+    """
+    fi = {"id": "test_id", "file_name": "test_file_name"}
+    copied_objects = {
+        "tcga-controlled/test_id/test_file_name": {
+            "id": "test_id",
+            "file_name": "test_file_name",
+        }
+    }
+    assert scripts.aws_replicate.is_changed_acl_object(fi, copied_objects, "tcga-open")
+
+    copied_objects = {
+        "ccle-open-access/test_id/test_file_name": {
+            "id": "test_id",
+            "file_name": "test_file_name",
+        }
+    }
+    assert scripts.aws_replicate.is_changed_acl_object(
+        fi, copied_objects, "gdc-ccle-controlled"
+    )
+
+
+@patch('scripts.indexd_utils.PROJECT_ACL', PROJECT_ACL)
+def test_update_url_with_new_acl(reset_records):
+    """
+    Test that updates indexd with new acl
+    """
+    mock_client = MockIndexdClient()
+    fi = {
+        "project_id": "TCGA-PNQS",
+        "id": "uuid1",
+        "file_name": "filename1",
+        "acl": "['phs000218',  'phs000219']",
+    }
+    indexd_utils.update_url(fi, mock_client)
+    doc = mock_client.get("uuid1")
+    assert doc.acl == ['phs000218', 'phs000219']
+
+
+@patch('scripts.indexd_utils.PROJECT_ACL', PROJECT_ACL)
+def test_update_url_with_new_url(reset_records):
+    """
+    Test that update indexd with new url
+    """
+    mock_client = MockIndexdClient()
+    fi = {
+        "project_id": "TCGA-PNQS",
+        "id": "uuid1",
+        "file_name": "filename1",
+        "acl": "['phs000218',  'phs000219']",
+        "url": "tcga-controlled/uuid1/filename1"
+    }
+    indexd_utils.update_url(fi, mock_client)
+    doc = mock_client.get("uuid1")
+    assert doc.urls == ['s3://tcga-controlled/uuid1/filename1']
+
+
+@patch('scripts.indexd_utils.PROJECT_ACL', PROJECT_ACL)
+def test_update_url_with_new_url2(reset_records):
+
```

Comment:
can you add a docstring to explain the difference between this test and the previous one